src/parsers/containers/isobmff/get_box.ts

```diff
@@ -113,9 +117,62 @@ function getUuidContent(
   }
 }
 
+/**
+ * For the next encountered box, return byte offsets corresponding to:
+ *   1. the starting byte offset for the next box (should always be equal to
+ *       `0`).
+ *   2. The beginning of the box content - meaning the first byte after the
+ *      size and the name of the box.
+ *   3. The first byte after the end of the box, might be equal to `buf`'s
+ *      length if we're considering the last box.
+ *
+ * `null` if no box is found.
+ * @param {Uint8Array} buf - the isobmff structure
+ * @param {Number} boxName - the 4-letter 'name' of the box as a 4 bit integer
+ * generated from encoding the corresponding ASCII in big endian.
+ */
+function getNextBoxOffsets(buf : Uint8Array
```

Comment:
This function seems really strange for me.
I don't see the point of having the first return tuple element being always the same value. Why returning it ?
And does it really means something the return the first byte ? It seems to be used to get the version number for CENC box but why having a more abstract function that seems to return some random info about a box ?

Reply:
For all these questions, it becomes clearer after reading the PR that this branch was previously based on: #784.

`getNextBoxOffsets` is a generic utils function for ISOBMFF parsing which works the same way than `getBoxOffsets` does in #784 with an exception: `getBoxOffsets` there returns a 3-tuple `[start, content_start, end[` for a specific box (based on its name) whereas `getNextBoxOffsets` returns that same tuple for the first encountered box. 

>  I don't see the point of having the first return tuple element being always the same value. Why returning it ?

The fact that it's always the same value is a little incidental.

The first element of that tuple is the starting byte of the next encountered box. For an ISOBMFF buffer to be valid and "parse-able", it needs to start with a box. Thus, this value will always be equal to `0`.

To me, it still makes sense because again, this is a generic utils function which is a variant of the `getBoxOffsets` API, optimized for when you know that you want offsets for the next box.

> why having a more abstract function that seems to return some random info about a box

Even a less "abstract" function which would just return the version of that box would have to perform the same logic

Because technically any box size can be defined on 8 bytes instead of 4, the offset where the `version` property of a PSSH box is can change.
To find out that offset, either we have to put that logic inline in every place that needs it, or we can provide some utils function which does the task for us.

That's where `getBoxOffsets` (in #784) and `getNextBoxOffsets` come into play.